src/theory/quantifiers/extended_rewrite.cpp

```diff
@@ -1563,6 +1585,571 @@ Node ExtendedRewriter::extendedRewriteArith(Node ret)
+    unsigned csize = bv::utils::getSize(a);
+    for (unsigned i = 0; i < csize; i++)
+    {
+      if (bv::utils::getBit(a, i) != bv::utils::getBit(b, i))
+      {
+        if (bv::utils::getBit(b, i))
+        {
+          return 0;
+        }
+        else
+        {
+          curr_ret = max_ret;
+        }
+      }
+    }
+  }
+
+  if (curr_ret == max_ret)
+  {
+    return max_ret;
+  }
+
+  Kind ak = a.getKind();
+  Kind bk = b.getKind();
+
+  //------------------recurse on CONCAT
+  if (ak == BITVECTOR_CONCAT || bk == BITVECTOR_CONCAT)
+  {
+    unsigned counter = bv::utils::getSize(a) - 1;
+    bool reca = ak == BITVECTOR_CONCAT;
+    Node cterm = reca ? a : b;
+    Node oterm = reca ? b : a;
+    int ccurr = 1;
+    // each piece must subsume
+    for (const Node& ctermc : cterm)
+    {
+      unsigned csize = bv::utils::getSize(ctermc);
+      Assert((counter + 1) >= csize);
+      Node extract =
+          bv::utils::mkExtract(oterm, counter, (counter + 1) - csize);
+      extract = Rewriter::rewrite(extract);
+
+      // check subsume
+      int cret = bitVectorSubsume(
+          reca ? ctermc : extract, reca ? extract : ctermc, strict);
+      if (cret == 0)
+      {
+        ccurr = 0;
+        break;
+      }
+      else if (cret > ccurr)
+      {
+        ccurr = cret;
+      }
+      if (counter >= csize)
+      {
+        counter = counter - csize;
+      }
+    }
+    // if all pieces subsume, then we take the max and update the return value
+    if (ccurr == max_ret)
+    {
+      return max_ret;
+    }
+    else if (ccurr > curr_ret)
+    {
+      curr_ret = ccurr;
+    }
+  }
+
+  //---------------recurse BITVECTOR_NOT
+  if (ak == BITVECTOR_NOT && bk == BITVECTOR_NOT)
+  {
+    return bitVectorSubsume(b[0], a[0], strict);
+  }
+
+  //--------------recurse extract
+  if (ak == BITVECTOR_EXTRACT && bk == BITVECTOR_EXTRACT
+      && a.getOperator() == b.getOperator())
+  {
+    return bitVectorSubsume(a[0], b[0], strict);
+  }
+
+  //---------------recurse BITVECTOR_AND/OR
+  bool reca = false;
+  Node rec;
+  if (ak == BITVECTOR_OR)
+  {
+    rec = a;
+    reca = true;
+  }
+  else if (bk == BITVECTOR_AND)
+  {
+    rec = b;
+  }
+  if (!rec.isNull())
+  {
+    for (const Node& rc : rec)
+    {
+      int cret = bitVectorSubsume(reca ? rc : a, reca ? b : rc, strict);
+      if (cret == max_ret)
+      {
+        return max_ret;
+      }
+      else if (cret > curr_ret)
+      {
+        curr_ret = cret;
+      }
+    }
+    rec = Node::null();
+  }
+
+  //---------------recurse ITE
+  if (ak == ITE)
+  {
+    rec = a;
+    reca = true;
+  }
+  else if (bk == ITE)
+  {
+    rec = b;
+    reca = false;
+  }
+  if (!rec.isNull())
+  {
+    int r1 = bitVectorSubsume(reca ? a[1] : a, reca ? b : b[1], strict);
+    if (r1 > curr_ret)
+    {
+      int r2 = bitVectorSubsume(reca ? a[2] : a, reca ? b : b[2], strict);
+      if (r2 > curr_ret)
+      {
+        curr_ret = r2 > r1 ? r1 : r2;
+        if (curr_ret = max_ret)
```

Comment:
This should be `curr_ret == max_reg`, no?

Reply:
Yes, ill fix